pulumi/infra/config.py

```
@@ -0,0 +1,162 @@
+  return overrides
+
+
+settings.managed_repo_overrides = _load_managed_repo_overrides(cfg.get_object("managedRepositories"))
+
+
+def _sanitize_bucket_component(value: str, label: str) -> str:
+  """Return a DNS-safe S3 bucket component derived from user input."""
+  normalized = value.strip().lower()
+
+  if _IPV4_PATTERN.match(normalized):
+    raise ValueError(f"{label} cannot be an IPv4 address.")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)